client/signup/steps/p2-site/index.jsx

```diff
@@ -296,6 +351,19 @@ class P2Site extends React.Component {
 		} );
 	};
 
+	handleSubdomainSuggestionClick = ( suggestedSubdomain ) => {
+		const site = suggestedSubdomain.substring( 0, suggestedSubdomain.indexOf( '.' ) );
+
+		this.formStateController.handleFieldChange( {
+			name: 'site',
+			value: site,
+		} );
+
+		this.setState( {
+			suggestedSubdomains: [],
+		} );
```

Comment:
I wonder if the user would not expect the form to be automatically submitted when they click one of those suggestions? I imagine we'd still fire the validation, so in the rare possibility that the subdomain is somehow taken, we'll have handle that. But in most cases the user will get taken to the next step (where they can change their selection still if needed).

Reply:
Interesting idea, cc @evilluendas. I personally would mind if it auto-submitted as I would want to verify all the details were set correctly.